File: cinema_app/lib/models/film.dart
class Film {
  final int id;
  final String title;
  final String description;
  final String type;
  final String director;
  final String imageUrl;
  final int likes;

  Film({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.director,
    required this.imageUrl,
    required this.likes,
  });

  factory Film.fromJson(Map<String, dynamic> json) {
    return Film(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      type: json['type'],
      director: json['director'],
      imageUrl: json['imageUrl'],
      likes: json['likes'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'type': type,
      'director': director,
      'imageUrl': imageUrl,
      'likes': likes,
    };
  }
}

File: cinema_api/Film.php
<?php
class Film {
    private $pdo;

    public function __construct($pdo) {
        $this->pdo = $pdo;
    }

    public function getAllFilms() {
        $stmt = $this->pdo->query("SELECT * FROM films");
        return $stmt->fetchAll();
    }

    public function getFilm($id) {
        $stmt = $this->pdo->prepare("SELECT * FROM films WHERE id = ?");
        $stmt->execute([$id]);
        return $stmt->fetch();
    }

    public function addFilm($title, $description, $type, $director, $imageUrl) {
        $stmt = $this->pdo->prepare("INSERT INTO films (title, description, type, director, imageUrl) VALUES (?, ?, ?, ?, ?)");
        $stmt->execute([$title, $description, $type, $director, $imageUrl]);
        return $this->getFilm($this->pdo->lastInsertId());
    }

    public function likeFilm($id) {
        $stmt = $this->pdo->prepare("UPDATE films SET likes = likes + 1 WHERE id = ?");
        $stmt->execute([$id]);
        return $this->getFilm($id);
    }

    public function updateFilm($id, $data) {
        $stmt = $this->pdo->prepare("UPDATE films SET title = ?, description = ?, type = ?, director = ?, imageUrl = ? WHERE id = ?");
        $stmt->execute([$data['title'], $data['description'], $data['type'], $data['director'], $data['imageUrl'], $id]);
        return $this->getFilm($id);
    }

    public function deleteFilm($id) {
        $stmt = $this->pdo->prepare("DELETE FROM films WHERE id = ?");
        return $stmt->execute([$id]);
    }
}
?>